Sources/Components/CollapseView/CollapseView.swift

```diff
@@ -0,0 +1,188 @@
+//
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import UIKit
+
+public protocol CollapseViewDelegate: AnyObject {
+    func didExpand(_ view: CollapseView)
+    func didCollapse(_ view: CollapseView)
+}
+
+public class CollapseView: UIView {
+    // MARK: Public properties
+
+    public weak var delegate: CollapseViewDelegate?
+
+    // MARK: Private properties
+
+    private enum State {
+        case expanded
+        case collapsed
+    }
+
+    private lazy var hairlineView: UIView = {
+        let view = UIView(withAutoLayout: true)
+        view.backgroundColor = .bgPrimary
+        view.dropShadow(color: .black, opacity: 0.6, offset: CGSize(width: 0, height: 0.9), radius: 1.2)
+        return view
+    }()
+
+    private lazy var selectorTitleView: SelectorTitleView = {
+        let view = SelectorTitleView(withAutoLayout: true)
+        view.updateButtonColor(.textSecondary, buttonDisabledColor: .textSecondary)
+        view.delegate = self
+        return view
+    }()
+
+    private lazy var contentView: UIView = {
+        let view = UIView(withAutoLayout: true)
+        view.backgroundColor = .bgPrimary
+        return view
+    }()
+
+    private var state: State
+
+    private let defaultHeight: CGFloat = 32.0
+    private let defaultContentViewHeight: CGFloat = 32.0
+
+    private var heightAnchorConstraint: NSLayoutConstraint?
+
+    // MARK: Parameters
+
+    private var collapsedTitle: String
+    private var expandedTitle: String
+    private var injectedView: UIView?
+    private var injectedViewHeight: CGFloat = 0.0
+
+    // MARK: Initalizer
+
+    public init(collapsedTitle: String, expandedTitle: String, presentViewInExpandedState: UIView?, heightOfView: CGFloat, withAutoLayout: Bool = false) {
+        self.collapsedTitle = collapsedTitle
+        self.expandedTitle = expandedTitle
+        self.injectedView = presentViewInExpandedState
+        self.injectedViewHeight = heightOfView
+        self.state = .collapsed
+
+        super.init(frame: .zero)
+
+        translatesAutoresizingMaskIntoConstraints = !withAutoLayout
+        backgroundColor = .bgPrimary
+
+        selectorTitleView.title = collapsedTitle
+        selectorTitleView.arrowDirection = .up
+
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    public func presentViewInExpandedState(_ view: UIView, heightOfView: CGFloat) {
```

Comment:
Is there any particular reason why we could not replace the view whenever we want? (Rather than just when it is expanded?). 

`viewToPresent(_ view---` ???